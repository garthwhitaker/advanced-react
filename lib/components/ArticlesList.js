import React from "react";
import PropTypes from "prop-types";
import Article from "./Article";

const ArticlesList = props => {
  return (
    <React.Fragment>
      {Object.values(props.articles).map(article => {
        return (
          <Article
            key={article.id}
            article={article}
            actions={props.articleActions}
          />
        );
      })}
    </React.Fragment>
  );
};

ArticlesList.propTypes = {
  articles: PropTypes.object,
  articleActions: PropTypes.object
};

export default ArticlesList;
