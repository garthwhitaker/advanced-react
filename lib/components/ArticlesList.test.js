import React from "react";
import ArticlesList from "./ArticlesList";

import renderer from "react-test-renderer";

describe("ArticlesList", () => {
  const props = {
    articles: { a: { id: "id" } },
    articleActions: { lookupAuthor: jest.fn(() => ({})) }
  };

  it("should render", () => {
    const tree = renderer.create(<ArticlesList {...props} />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
