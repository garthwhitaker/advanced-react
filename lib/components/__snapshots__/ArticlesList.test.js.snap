// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticlesList should render 1`] = `
Array [
  <div
    style={
      Object {
        "fontSize": 14,
        "fontWeight": "bold",
      }
    }
  />,
  <div
    style={
      Object {
        "color": "#c1c1c1",
        "fontSize": 11,
      }
    }
  >
    Invalid Date
  </div>,
  <a>
     
  </a>,
  <div
    style={
      Object {
        "padding": 5,
      }
    }
  />,
  <hr />,
]
`;
